packages/core/admin/admin/src/content-manager/components/DynamicTable/BulkActionsBar/index.js

```diff
@@ -214,34 +214,46 @@ const BulkActionsBar = ({
   };
 
   const handleBulkDelete = async () => {
+    let result = null;
+
     try {
       setIsConfirmButtonLoading(true);
-      await onConfirmDeleteAll(selectedEntries);
+      result = await onConfirmDeleteAll(selectedEntries);
     } catch (err) {
       // The error is handled in content-manager/pages/ListView/index.js
       // handleConfirmDeleteAllData catch block
     } finally {
       setIsConfirmButtonLoading(false);
       toggleDeleteModal();
+    }
+
+    if (result) {
       clearSelectedEntries();
     }
   };
 
   const handleBulkPublish = async () => {
+    let result = null;
+
     try {
       setIsConfirmButtonLoading(true);
-      await onConfirmPublishAll(selectedEntries);
+      result = await onConfirmPublishAll(selectedEntries);
     } catch (error) {
       // The error is handled in content-manager/pages/ListView/index.js
       // bulkPublishMutation onError callback
     } finally {
       setIsConfirmButtonLoading(false);
       togglePublishModal();
+    }
+
+    if (result) {
```

Comment:
Since this won't work as we wanted you could just do:

```js
try {
      setIsConfirmButtonLoading(true);
      await confirmAction(selectedEntries);
      setIsConfirmButtonLoading(false);
      toggleModal();
			clearSelectedEntries()
    } catch (error) {
      setIsConfirmButtonLoading(false);
      toggleModal();
    }
```

Which avoids what you've done now :) It's duplication, but because of how the DynamicTable is written, this is the only way imo.

Reply:
Man, I swore I tried that! This is much better than what I have currently, thanks. updated.

Reply:
I haven't tested this worked, i just assumed lol